### ISSUE TITLE
Using reference comparison for value type - problem

### ISSUE THREAD

**Issue by lontivero** [NicolasDorier/NBitcoin#97]
@NicolasDorier, LockTime is a value type and that means it can never be **null**. For that same reason (LockTime is not a reference type) the reference comparison is always **false**

 Please check the following code:

```c#
		public override bool Equals(object obj)
		{
			LockTime item = obj is LockTime ? (LockTime)obj : default(LockTime);
			if(item == null)       //  <----------- HERE (always false)
				return false;
			return _value.Equals(item._value);
		}
		public static bool operator ==(LockTime a, LockTime b)
		{
			if(System.Object.ReferenceEquals(a, b))  // <------- HERE (this is always false)
				return true;
			if(((object)a == null) || ((object)b == null)) //  <----------- HERE (always false)
				return false;     
			return a._value == b._value;
		}
```

**Comment by NicolasDorier** [comment]
nice catch. At least the bug does not break the implementation.

**Comment by NicolasDorier** [comment]
fixed in f13592fe
